osquery/tables/system/posix/authorized_keys.cpp

```diff
@@ -48,13 +68,78 @@ void genSSHkeysForUser(const std::string& uid,
       logger.log(google::GLOG_WARNING, s.getMessage());
       continue;
     }
+
     // Protocol 1 public key consist of: options, bits, exponent, modulus,
     // comment; Protocol 2 public key consist of: options, keytype,
     // base64-encoded key, comment.
     for (const auto& line : split(keys_content, "\n")) {
       if (!line.empty() && line[0] != '#') {
-        Row r = {{"uid", uid}, {"key", line}, {"key_file", keys_file.string()}};
-        r["pid_with_namespace"] = "0";
+        auto splitted_line = split(line, " ");
+        if (splitted_line.empty()) {
+          // Line syntax is invalid.
+          continue;
+        }
+
+        auto splitted_line_part = splitted_line.begin();
+        for (; splitted_line_part != splitted_line.end();
+             splitted_line_part++) {
+          if (find(kSSHKeyTypes.begin(),
+                   kSSHKeyTypes.end(),
+                   *splitted_line_part) != kSSHKeyTypes.end()) {
+            break;
+          }
+        }
+
+        // The current line does not contain key type.
+        if (splitted_line_part == splitted_line.end()) {
+          // rest of the line after 'zos-key-ring-label' should be ignored.
+          if (isKeyRingLabelOptExist(line)) {
```

Comment:
this is called with `line`, why is it inside the for loop over splitted_line?

Reply:
I've fixed the code according to this comment.